src/template/wasm2c_simd.declarations.c

```diff
@@ -2,6 +2,9 @@
 #define SIMD_FORCE_READ(var) __asm__("" ::"x"(var));
 #elif defined(__GNUC__) && defined(__aarch64__)
 #define SIMD_FORCE_READ(var) __asm__("" ::"w"(var));
+#elif WASM_RT_MEMCHECK_GUARD_PAGES
+// best-effort using volatile
+#define SIMD_FORCE_READ(var) (void)*(volatile v128*)&var;
```

Comment:
Why only with `WASM_RT_MEMCHECK_GUARD_PAGES`, why not make this the default fallback?

Reply:
Can we use `_asm__("" ::"x"(var));` on x86 too maybe?

Reply:
I think we don't want it as the default fallback always because:
1. We don't need this force_read operation when using bounds checks
2. It may have some performance implications

Reply:
Yes, I would expect SIMD_FORCE_READ to not be used at all if its not needed. 

Defining it to something that doesn't actually force anything seems wrong.

Reply:
SIMD_FORCE_READ_IF_NECESSARY?

Reply:
Lets just leave the name as is.

I do this think we should not define it to anything unless `WASM_RT_MEMCHECK_GUARD_PAGES` is set though.

Reply:
that would make the code way messier tho... hmm...

SIMD_CHECK_PAGE_FAULT? we already have bounds-check macros that do nothing when they're not needed...